src/app/src/App.js

```diff
@@ -1,18 +1,64 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { bool } from 'prop-types';
+import { bool, func, string } from 'prop-types';
+import IdleTimer from 'react-idle-timer';
 
 import Navbar from './Navbar';
 import Screensaver from './Screensaver';
 import GlobalStyle from './globalStyle';
+import { resetAppState } from './actions';
+import { RESET, PAUSE } from './constants';
 
 class App extends Component {
+    constructor() {
+        super();
+        this.idleTimer = null;
+        this.resetTimer = this.resetTimer.bind(this);
+        this.pauseTimer = this.pauseTimer.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps) {
```

Comment:
Perhaps this is a lack of familiarity with React, but does this need to be in `shouldComponentUpdate`?

We always return `true` here, so can't we use `componentDidUpdate` instead?

Reply:
Good suggestion!